src/core/core.controller.js

```diff
@@ -70,8 +75,13 @@ function mergeScaleConfig(/* objects ... */) {
 	});
 }
 
-function mergeConfig(/* objects ... */) {
-	return helpers.merge(helpers.clone(arguments[0]), [].slice.call(arguments, 1), {
+/**
+ * Recursively merge the given config objects as the root options by handling
+ * default scale options for the `scales` and `scale` properties, then returns
+ * a deep copy of the result, thus doesn't alter inputs.
+ */
+function mergeConfig(/* config objects ... */) {
+	return helpers.merge({}, Array.prototype.slice.call(arguments), {
```

Comment:
I think using `[].slice.call` would save 13 bytes minified. (performance is in par: [jsperf](https://jsperf.com/array-prototype-slice-call-vs-slice-call))

Reply:
Based on the [comment for the MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments#Description), it prevents to allocate an empty array using the prototype approach.

Reply:
Done